clippy_lints/src/macro_use.rs

```diff
@@ -213,3 +213,7 @@ impl<'tcx> LateLintPass<'tcx> for MacroUseImports {
         }
     }
 }
+
+fn in_macro(span: Span) -> bool {
+    span.from_expansion() && !matches!(span.ctxt().outer_expn_data().kind, ExpnKind::Desugaring(..))
```

Comment:
Ah, so this is the only place where we can't get rid of the distinction between macro and desugaring, yet.

Reply:
Right. Actually this lint probably should look for `ExpnKind::Macro(MacroKind::Bang`